src/server/routes/contract/events/addContractSubscription.ts

```diff
@@ -0,0 +1,90 @@
+import { Static, Type } from "@sinclair/typebox";
+import { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import { upsertChainIndexer } from "../../../../db/chainIndexers/upsertChainIndexer";
+import { upsertContractSubscription } from "../../../../db/contractSubscriptions/createContractSubscription";
+import { getContractSubscriptionsUniqueChainIds } from "../../../../db/contractSubscriptions/getContractSubscriptions";
+import { getSdk } from "../../../../utils/cache/getSdk";
+import { logger } from "../../../../utils/logger";
+import {
+  contractParamSchema,
+  standardResponseSchema,
+} from "../../../schemas/sharedApiSchemas";
+import { getChainIdFromChain } from "../../../utils/chain";
+
+const responseSchema = Type.Object({
+  result: Type.Object({
+    chain: Type.String(),
+    contractAddress: Type.String(),
+    status: Type.String(),
+  }),
+});
+
+responseSchema.example = {
+  result: {
+    chain: "ethereum",
+    contractAddress: "0x....",
+    status: "success",
+  },
+};
+
+export async function addContractSubscription(fastify: FastifyInstance) {
+  fastify.route<{
+    Params: Static<typeof contractParamSchema>;
+    Reply: Static<typeof responseSchema>;
+  }>({
+    method: "POST",
+    url: "/contract/:chain/:contractAddress/events/subscribe",
+    schema: {
+      summary: "Subscribe to contract events",
+      description: "Subscribe to contract events",
+      tags: ["Contract", "Index"],
+      operationId: "write",
+      params: contractParamSchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: responseSchema,
+      },
+    },
+    handler: async (request, reply) => {
+      const { chain, contractAddress } = request.params;
```

Comment:
are we supporting all chains?? if `Yes` then ignore, if not then we need to have an allowList

Reply:
all chains